crates/oxc_linter/src/rules/no_mixed_operators.rs

```diff
@@ -0,0 +1,344 @@
+        // never enter the code path.
+    }
+
+    /// Report mixed operator pare between node and parent corresponding to configuration.
+    fn report(&self, node: AstKind, parent: AstKind, ctx: &LintContext<'_>) {
+        let (node_operator, node_left_span, node_right_span) = match node {
+            AstKind::BinaryExpression(expr) => {
+                (expr.operator.as_str(), expr.left.span(), expr.right.span())
+            }
+            AstKind::LogicalExpression(expr) => {
+                (expr.operator.as_str(), expr.left.span(), expr.right.span())
+            }
+            _ => unreachable!(),
+        };
+        // Since we don't store the exact span of the operators, approximate that span to be between the lhs
+        // and rhs of the expression.
+        let node_operator_span = Span::new(node_left_span.end + 1, node_right_span.start - 1);
+
+        let (parent_operator, parent_left_span, parent_right_span) = match parent {
+            AstKind::BinaryExpression(expr) => {
+                (expr.operator.as_str(), expr.left.span(), expr.right.span())
+            }
+            AstKind::LogicalExpression(expr) => {
+                (expr.operator.as_str(), expr.left.span(), expr.right.span())
+            }
+            AstKind::ConditionalExpression(expr) => {
+                // For conditional operators, the span covers both ? and :
+                ("?:", expr.test.span(), expr.alternate.span())
+            }
+            _ => unreachable!(),
+        };
+        let parent_operator_span = Span::new(parent_left_span.end + 1, parent_right_span.start - 1);
+
+        let (node_operator, node_precedence) = operator_and_precedence(node_operator).unwrap();
+        let (parent_operator, parent_precedence) =
+            operator_and_precedence(parent_operator).unwrap();
+        if self.in_the_same_group(node_operator, parent_operator)
+            && !(self.allow_same_precedence && node_precedence == parent_precedence)
+        {
+            // Report error at both operators
+            ctx.diagnostic(NoMixedOperatorsDiagnostic(
+                node_operator,
+                parent_operator,
+                node_operator_span,
+            ));
+            ctx.diagnostic(NoMixedOperatorsDiagnostic(
+                node_operator,
+                parent_operator,
+                parent_operator_span,
+            ));
+        }
+    }
+
+    fn in_the_same_group(&self, op1: &str, op2: &str) -> bool {
+        self.groups.iter().any(|group| {
+            let mut contains_op1 = false;
+            let mut contains_op2 = false;
+            for &op in group {
+                if op == op1 {
+                    contains_op1 = true;
+                }
+                if op == op2 {
+                    contains_op2 = true;
+                }
+            }
+            contains_op1 && contains_op2
+        })
+    }
+}
+
+impl Default for NoMixedOperators {
```

Comment:
It is preferred to put these builtin `impl`s next to the struct, so it's easier to read and find.

Reply:
Done.